src/extension.ts

```diff
@@ -103,3 +111,70 @@ async function workWithServer(verb: string, callback: ServerFunction): Promise<v
 export function deactivate() {
     nisabaLogger.stopLogging();
 }
+
+async function getSymbols(document: vscode.TextDocument): Promise<vscode.SymbolInformation[]> {
```

Comment:
I don't think this needs to be async, as it can directly return a `SymbolInformation` (or `DocumentSymbol`) array, without wrapping it in a `Promise`? (judging from the [`ProviderResult` examples](https://code.visualstudio.com/api/references/vscode-api#ProviderResult%3CT%3E))